src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -54,24 +54,58 @@ public async Task<ImmutableArray<CodeAction>> AddConstructorParametersFromMember
             }
         }
 
+        /// <summary>
+        /// Try to find a constructor in <paramref name="containingType"/> whose parameters is the subset of <paramref name="parameters"/> by comparing name.
+        /// If multiple constructors meet the condition, the one with more parameters will be returned.
+        /// It will not consider those constructors as potential candidates
+        /// 1. Constructor with empty parameter list.
+        /// 2. Constructor's parameter list contains 'ref' or 'params'
+        /// </summary>
+        protected override IMethodSymbol GetDelegatedConstructor(
+            INamedTypeSymbol containingType,
+            ImmutableArray<IParameterSymbol> parameters)
+        {
+            return containingType.InstanceConstructors
+                .WhereAsArray(constructor => IsParamtersContainedInConstructor(constructor, parameters.SelectAsArray(p => p.Name)))
+                .OrderByDescending(constructor => constructor.Parameters.Length)
+                .FirstOrDefault();
+        }
+
+        private bool IsParamtersContainedInConstructor(
+            IMethodSymbol constructor,
+            ImmutableArray<string> parametersName)
+        {
+            var constructorParams = constructor.Parameters;
+            if (constructorParams.Length > 0
+                && constructorParams.All(parameter => parameter.RefKind == RefKind.None)
+                && !constructorParams.Any(p => p.IsParams))
+            {
+                return parametersName.Except(constructorParams.Select(p => p.Name)).Any();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
         private IEnumerable<CodeAction> CreateCodeActions(Document document, State state)
         {
             var lastParameter = state.DelegatedConstructor.Parameters.Last();
             if (!lastParameter.IsOptional)
             {
-                yield return new AddConstructorParametersCodeAction(this, document, state, state.Parameters);
+                yield return new AddConstructorParametersCodeAction(this, document, state, state.MissingParameters);
             }
 
-            var parameters = state.Parameters.Select(p => CodeGenerationSymbolFactory.CreateParameterSymbol(
+            var missingOptionalParameters = state.MissingParameters.SelectAsArray(p => CodeGenerationSymbolFactory.CreateParameterSymbol(
                 attributes: default,
                 refKind: p.RefKind,
                 isParams: p.IsParams,
                 type: p.Type,
                 name: p.Name,
                 isOptional: true,
-                hasDefaultValue: true)).ToList();
+                hasDefaultValue: true));
 
-            yield return new AddConstructorParametersCodeAction(this, document, state, parameters);
+            yield return new AddConstructorParametersCodeAction(this, document, state, missingOptionalParameters);
```

Comment:
you call it missingParameters inside the constructor, but missingOptionalParameters here.  can you make the names consistent?